crates/bevy_ecs/src/system/function_system.rs

```diff
@@ -644,6 +650,22 @@ impl<Param: SystemParam> SystemState<Param> {
         self.meta.last_run = change_tick;
         param
     }
+
+    /// Returns a reference to the current system param states.
+    pub fn param_state(&self) -> &Param::State {
+        &self.param_state
+    }
+
+    /// Returns a mutable reference to the current system param states.
+    /// Marked as unsafe because modifying the system states may result in violation to certain
+    /// assumptions made by the [`SystemParam`]. Use with care.
+    /// 
+    /// Safety: Modifying the system param states may have unintented consequences.
+    /// the param state is generally considered to be owned by the [`SystemParam`]. Modifications
+    /// should respect any invariants as required by the [`SystemParam`].
```

Comment:
It might be helpful to provide an example of when this might be the case, as I'm not sure why it needs to be unsafe.

Reply:
For example, modifying the system state of `ResMut` without also updating [`SystemMeta::component_access_set`] will obviously create issues.